projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -1666,6 +1677,62 @@ protected function filter_non_public_posts( array $related_posts ) {
 		return $filtered;
 	}
 
+	/**
+	 * Generates a context for the related content (second line in related post output).
+	 * Order of importance:
+	 *   - First category (Not 'Uncategorized')
+	 *   - First post tag
+	 *   - Number of comments
+	 *
+	 * @param int $post_id - the post ID.
+	 * @uses get_the_category, get_the_terms, get_comments_number, number_format_i18n, __, _n
+	 * @return string
+	 */
+	protected function generate_related_post_context_block( $post_id ) {
```

Comment:
What's the rationale for making this new function instead of updating the existing `generate_related_post_context` function?

Is it possible to update the original context to include the link data?

Reply:
This is mostly the same as my comment above but the main reasoning is to keep the legacy code path working as is with the ability to filter the results via PHP code snippets.

Then we have the new "block context" for use in the block editor where I don't think we really want a user-filtered value being pushed into the editor. My thinking is the context should work much like the date. If enabled, we generate a date for display. We don't enable any hooks to allow the user to then modify how the date is formatted before we output to screen.

In the legacy code path we have some hooks that allow the user to modify the context as they see fit. They can use this to add a timestamp, author, tag cloud, etc before the final string is added to the page HTML. It feels like something we should preserve but I'm not sure we want to bring it into the block editor.

> Is it possible to update the original context to include the link data?

To actually answer this question, yes, we could update the original context but we'd be modifying the "input" to the user-supplied filter function which could result in breakage, depending on how defensively coded their PHP snippets are.

This is all with the caveat that this is a bit of a legacy scenario. The original context would only be seen (and user-supplied filters run) on sites using themes that haven't been updated for FSE.

Reply:
I agree with your approach of not altering non-FSE Related Posts widget behavior.